### PR TITLE
boards/nucleo144-f303: initial support

### PR DESCRIPTION
**Pull request by aabadie**
Another port of Nucleo board. This one is for the [Nucleo-F303ZE](http://www.st.com/en/evaluation-tools/nucleo-f303ze.html).

~~Let's wait for the CI to fix the usual memory overflow issues.~~

### CODE REVIEW

**Comment by kYc0o** [comment]
?? and why?

**Comment by aabadie** [comment]
oups, I deleted the wrong branch...

File: boards/nucleo144-f303/Makefile.features [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Put defined MCU peripherals here (in alphabetical order)
+FEATURES_PROVIDED += periph_cpuid
+FEATURES_PROVIDED += periph_gpio
+FEATURES_PROVIDED += periph_spi
+FEATURES_PROVIDED += periph_timer
+FEATURES_PROVIDED += periph_uart
+FEATURES_PROVIDED += periph_pwm
```
Comment (kYc0o) [inline boards/nucleo144-f303/Makefile.features]: Remember the alphabetical order.

**Comment by aabadie** [comment]
@kYc0o : done and rebased

File: boards/nucleo144-f303/include/periph_conf.h [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    {
+        .dev      = TIM2,
+        .max      = 0xffffffff,
+        .rcc_mask = RCC_APB1ENR_TIM2EN,
+        .bus      = APB1,
+        .irqn     = TIM2_IRQn
+    }
+};
+
+#define TIMER_0_ISR         isr_tim5
```
Comment (aabadie) [inline boards/nucleo144-f303/include/periph_conf.h]: the error is here, it should be `isr_tim2`

**Comment by aabadie** [comment]
@astralien3000, I forced push an update, It should be ok now. @kYc0o can you make a quick code review again ?

**Comment by astralien3000** [comment]
PWM 0 (TIM1) is not working

**Comment by astralien3000** [comment]
Everything seems ok (default, uart, pwm, timer)

**Comment by aabadie** [comment]
@kYc0o , can you ACK ?

**Comment by aabadie** [comment]
ping @kYcOo

**Comment by aabadie** [comment]
ping @kYc0o, for an ACK please :)

**Pull request merged**